win32ss/user/ntuser/keyboard.c

```diff
@@ -1020,8 +1020,9 @@ UserSendKeyboardInput(KEYBDINPUT *pKbdInput, BOOL bInjected)
         }
         else
         {
-            wVk = pKbdInput->wVk & 0xFF;
+            wVk = pKbdInput->wVk;
         }
+        wVk &= 0xFF;
```

Comment:
None of your testcases showcases this behavior, since you use a byte for you VK?

Reply:
The difference this change is making is reflected when the `if (pKbdInput->dwFlags & KEYEVENTF_SCANCODE)` is executed. Since the else block also had the same operating being done, I basically clubbed both in the ANDings into a single line. Should I change it to AND only in the if block?

Reply:
I also want to mention that in the case of extended keys with `KEYEVENTF_SCANCODE` flag, the `IntVscToVk` does in fact return a word `E0xx`.

Reply:
I would in fact suggest keeping the `& 0xFF` in both branches in this case -- mainly so that you can put a descriptive comment above it in the `KEYEVENTF_SCANCODE` case that explains why it needs to be done there (in particular -- why does the lookup table contain values with the high bits set? Is that needed for a different code path, but not here?). You can also mention that E0xx example there.
If you have a good understanding of what's going on here, it could in fact be useful to add a comment to the non-scancode case as well, explaining why the masking is needed there.

Reply:
Correction: The word returned in the case mentioned is `0x01XX`, not `0xE000`.

Reply:
Yes, these high bits in the lookup table are required. `UserProcessKeyboardInput` uses this to set the `dwFlags`. 
![image](https://user-images.githubusercontent.com/22453292/133908701-16149e02-e3c7-4a22-ab0c-6bbc36f4d1cb.png)

One can see that, an ANDing action similar to my changes is performed here also, before sending the `wVk` to `UserSendKeyboardInput`.

Reply:
I think the `&= 0xFF` operation is done in both the cases for the same reason, to remove all the virtual key flags.

Reply:
If the explanation is the same, then keeping it outside the if makes sense to me.

Reply:
I have added the comment alongwith keeping it outside the if, apart from the other notes. Do let me know if something else is still left.